temporal-testing/src/main/java/io/temporal/testing/TestWorkflowEnvironmentInternal.java

```diff
@@ -150,7 +150,11 @@ public void close() {
     workerFactory.shutdownNow();
     workerFactory.awaitTermination(10, TimeUnit.SECONDS);
     service.close();
-    workflowServiceStubs.shutdown();
+    if (Boolean.parseBoolean(System.getenv("USE_DOCKER_SERVICE"))) {
+      workflowServiceStubs.shutdown();
+    } else {
+      workflowServiceStubs.shutdownNow();
+    }
```

Comment:
This change made a difference running builds locally, but doesn't seem to make any difference in Buildkite.

Reply:
We shouldn't do shutdownNow unless we want to drop running tasks. If all we want it to ensure that all inflight work is being processed before the shutdown, we should use `awaitTermination`.

Reply:
I propose to do shutdownNow only for in-memory tests. We do it after the test has already executed. So do you think it matters that we drop all the running tasks at that point? This way we don't have to explicitly set longer timeouts for a few tests.

Reply:
According to the documentation:
* shutdown() will just tell the executor service that it can't accept new tasks, but the already submitted tasks continue to run
* shutdownNow() will do the same AND will try to cancel the already submitted tasks by interrupting the relevant threads. Note that if your tasks ignore the interruption, shutdownNow will behave exactly the same way as shutdown

In my understanding, what we should try to achieve in the close method is semantics of shutdown + awaitTermination, which would not only reject new requests, but will also wait until all inflight requests finish. Otherwise there would be a chance that you call `close` but some requests are still inflight after `close` returns.

Reply:
Decided to land as is, because closing is done after the test. We will adjust accordingly in the future.